plugins/source/azure/client/helpers.go

```diff
@@ -22,3 +23,36 @@ func ParseResourceGroup(resourceID string) (string, error) {
 	}
 	return match[2], nil
 }
+
+type syncData struct {
+	data any
+	err  error
+	once *sync.Once
+}
+
+func loadOrStore(m *sync.Map, key string, f func() (any, error)) (any, error) {
+	temp, _ := m.LoadOrStore(key, syncData{
+		data: nil,
+		err:  nil,
+		once: &sync.Once{},
+	})
+	d := temp.(syncData)
+	if d.data == nil && d.err == nil {
+		d.once.Do(func() {
+			dt, err := f()
+			if err != nil {
+				dt = nil
+				d.once = &sync.Once{}
+			}
+			m.Store(key, syncData{
+				data: dt,
+				err:  err,
+				once: d.once,
+			})
+		})
+		temp, _ = m.Load(key)
```

Comment:
Modifying `d` inside `Do()` triggers the race detector for some reason (if ptr `*syncData` is used inside the map) so we just store and load it again.